VideoLocker/src/main/java/org/edx/mobile/model/api/CourseEntry.java

```diff
@@ -248,22 +252,72 @@ public boolean isGroupAvailable(SocialUtils.SocialType type) {
 
     }
 
-    public String getDescription(Context context){
+    @SuppressLint("SimpleDateFormat")
+    public String getDescription(Context context) {
         StringBuilder detailBuilder = new StringBuilder();
-        if ( getOrg() != null){
-            detailBuilder.append( getOrg());
+
+        if (getOrg() != null) {
+            detailBuilder.append(getOrg());
         }
-        if ( getNumber() != null) {
-            if (detailBuilder.length() > 0){
+
+        if (getNumber() != null) {
+            if (detailBuilder.length() > 0) {
                 detailBuilder.append(" | ");
             }
-            detailBuilder.append( getNumber());
+            detailBuilder.append(getNumber());
+
+        }
+
+        SimpleDateFormat dateFormat = new SimpleDateFormat("MMMM dd");
+        Date startDate = DateUtil.convertToDate(start);
+        String startDt;
+        Date endDate = DateUtil.convertToDate(end);
+        String endDt;
+        if (detailBuilder.length() > 0) {
+            detailBuilder.append(" | ");
+        }
+        if (hasDefaultStartDate()) {
+            detailBuilder.append(context.getString(R.string.assessment_empty_coming_soon).toUpperCase());
+        } else if (isStarted() && isEnded()) {
+            if (endDate != null) {
+                endDt = context.getString(R.string.label_ended)
+                        + " - " + dateFormat.format(endDate);
+                detailBuilder.append(endDt.toUpperCase());
+            }
+        } else if (isStarted() && !isEnded()) {
+            if (endDate != null) {
+                endDt = context.getString(R.string.label_ending_on)
+                        + " - " + dateFormat.format(endDate);
+                detailBuilder.append(endDt.toUpperCase());
+
+            }
+        } else {
+            if (startDate != null) {
+                startDt = context.getString(R.string.label_starting_from)
+                        + " - " + dateFormat.format(startDate);
+                detailBuilder.append(startDt.toUpperCase());
+            }
+        }
 
+        if (detailBuilder.length() > 0) {
+            int secondlastCharIndex = detailBuilder.length() - 2;
+            if (detailBuilder.charAt(secondlastCharIndex) == '|') {
+                detailBuilder.delete(secondlastCharIndex, detailBuilder.length());
+            }
         }
-        //https://openedx.atlassian.net/browse/MA-858
-        //we remove the ending data for now
 
         return detailBuilder.toString();
     }
 
+    /**
+     * This functions checks if a course's start date is set to default
+     * i.e. when its {@link #start_type} is set to {@link StartType#EMPTY}
+     *
+     * @return true if start date is default, false otherwise
+     */
+    @SuppressLint("SimpleDateFormat")
+    public boolean hasDefaultStartDate() {
```

Comment:
This method is not named appropriately anymore, and I don't think we need it at all now, as it's only being used in one place in this very class.